### PR TITLE
Spring 08

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/ua/com/cinema/dao/RoleDao.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ua.com.cinema.dao;
+
+import ua.com.cinema.model.Role;
+
+public interface RoleDao {
+    Role add(Role role);
+
+    Role getRoleByName(String roleName);
```
Comment (Sofasmile) [inline src/main/java/ua/com/cinema/dao/RoleDao.java]:
```suggestion
    Optional<Role> getRoleByName(String roleName);
```

File: src/main/java/ua/com/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(userOptional.get().getPassword());
+            builder.roles(userOptional.get().getRoles()
+                    .stream()
+                    .map(x -> x.getRoleName().name())
+                    .toArray(String[]::new));
+        } else {
+            throw new UsernameNotFoundException("User not found.");
+        }
+        return builder.build();
```
Comment (Sofasmile) [inline src/main/java/ua/com/cinema/security/CustomUserDetailsService.java]:
```suggestion
            return builder.build();
        }
        throw new UsernameNotFoundException("User not found.");
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job)

File: src/main/java/ua/com/cinema/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        user.setRoles(Set.of(roleService.getRoleByName("USER")));
+        user.setRoles(Set.of(roleService.getRoleByName("USER").get()));
```
Comment (Sofasmile) [inline src/main/java/ua/com/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
        user.setRoles(Set.of(roleService.getRoleByName("USER")));
```

File: src/main/java/ua/com/cinema/service/RoleService.java [hunk 4]
```diff
@@ -1,9 +1,10 @@
 package ua.com.cinema.service;
 
+import java.util.Optional;
 import ua.com.cinema.model.Role;
 
 public interface RoleService {
     void add(Role role);
 
-    Role getRoleByName(String roleName);
+    Optional<Role> getRoleByName(String roleName);
```
Comment (Sofasmile) [inline src/main/java/ua/com/cinema/service/RoleService.java]: in this case we should use Optional only on the dao layer

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**